### ISSUE TITLE
ReturnCount false positive with lambdas

### ISSUE THREAD

**Issue by maragues-kolibree** [arturbosch/detekt#1317]
## Expected Behavior
ReturnCount does not take into account `return` inside lambdas

## Observed Behavior
If you have multiple returns inside a lambda, ReturnCount is triggered

## Steps to Reproduce
This function violates ReturnCount, and I think it's a false positive
```kotlin
fun readUsers(name: String): Flowable<User> {
        return userDao.read(name)
            .flatMap {
                if (it.isEmpty()) return@flatMap Flowable.empty<User>()

                return@flatMap Flowable.just(it[0])
            }
    }
```

I know the last return can be removed and that a flowable doesn't make sense here, but let's ignore that

## Your Environment

* Version of detekt used: 1.0.0-RC10
* Version of Gradle used (if applicable): 4.9
* Operating System and version: MacOS 10.14

**Comment by arturbosch** [comment]
Agree with @schalkms .

**Comment by schalkms** [comment]
Closed with ee6a252

**Comment by arturbosch** [comment]
Woot, but it is not merged yet oO?

**Comment by 3flex** [comment]
https://github.com/arturbosch/detekt/commit/35686e849607b400307e82b55816cd50bb28cdc8 is in master... can close?

**Comment by schalkms** [comment]
Closed with 35686e8
